packages/s/soci/xmake.lua

```diff
@@ -8,24 +8,33 @@ package("soci")
 
     add_versions("v4.0.3", "4b1ff9c8545c5d802fbe06ee6cd2886630e5c03bf740e269bb625b45cf934928")
 
-    add_configs("empty", {description = "Build empty backend", default = false, type = "boolean"})
-    add_configs("sqlite3", {description = "Build sqlite3 backend", default = false, type = "boolean"})
-    add_configs("db2", {description = "Build db2 backend", default = false, type = "boolean"})
-    add_configs("odbc", {description = "Build odbc backend", default = false, type = "boolean"})
-    add_configs("oracle", {description = "Build oracle backend", default = false, type = "boolean"})
-    add_configs("firebird", {description = "Build firebird backend", default = false, type = "boolean"})
-    add_configs("mysql", {description = "Build mysql backend", default = false, type = "boolean"})
-    add_configs("postgresql", {description = "Build postgresql backend", default = false, type = "boolean"})
-    add_configs("boost", {description = "Build boost backend", default = false, type = "boolean"})
+    local backends = {
+        "empty",
+        "sqlite3",
+        "db2",
+        "odbc",
+        "oracle",
+        "firebird",
+        "mysql",
+        "postgresql",
+    }
+    for _, backend in ipairs(backends) do
+        add_configs(backend, {description = "Build " .. backend .. " backend", default = false, type = "boolean"})
+    end
+    add_configs("boost", {description = "Enable boost integration", default = false, type = "boolean"})
+    add_configs("visibility", {description = "Enable hiding private symbol using ELF visibility if supported by the platform", default = true, type = "boolean"})
 
-    add_deps("cmake")
+    add_deps("cmake <=3.20", {private = true})
```

Comment:
I mean patch the lower bound version at [this link](https://github.com/SOCI/soci/blob/master/CMakeLists.txt#L13), not in the xmake port file.

Reply:
I made the patch, the checksum works on windows but fails on every other platform.

Reply:
Probably because the checksum was made on a CRLF file, turn the line endings to LF and check it again